Trading bot

bonbon123123 opened the pull request.

amasend committed:
Merge pull request #16 from bonbon123123/downloader

Downloader

amasend committed:
Revert "Downloader"

amasend committed:
Merge pull request #17 from amasend/revert-16-downloader

Revert "Downloader"

bonbon123123 committed:
Added bot body with influxdb logs

bot managaer is empty as of now becouse nothing of value is ready

amasend commented on bot_management/bot_body.py:
```diff
@@ -0,0 +1,50 @@
+        self.bot_id = bot_id
+        self.time_to_live = time_to_live
+
+    def switcher(self, i="waiting"):
```
What switcher is for?

amasend reviewed (changes requested):
I am waiting for bot manager and tests.

bonbon123123 committed:
Returning deleted files

amasend commented on bot_management/bot_body.py:
```diff
@@ -0,0 +1,50 @@
+
+
+class Bot:
+    def __init__(self, bot_id:str, time_to_live):
```
What exactly is `time_to_live` here? And why we need to write it to DB?

amasend reviewed (changes requested):
I do not see any part that calls CCXT here, why? BOTs should make its own requests to an exchange.

bonbon123123 committed:
Added bot manager, tests, comments

Stopping bot isnt implemented yet

bonbon123123 committed:
added stop function to bot, comments and test

user now can change symbol of the exchange

amasend commented on bot_management/bot_body.py:
```diff
@@ -0,0 +1,126 @@
+from time import sleep, ctime
+import helpfull_functions
+from binance.client import Client
```
We were using `CCXT` as this is more general library to handle a range of different exchanges. But in this case to speed up your work we can use directly a `binance` packakge. But this is not recommended.

amasend commented on bot_management/bot_body.py:
```diff
@@ -0,0 +1,126 @@
+        self.bot_id = bot_id
+
+    def switcher(self, bot_id, binance_client, amount, what_to_do="waiting", test="0"):
+        """bot function that does all what bot does"""
```
What is this description about?

amasend commented on bot_management/bot_body.py:
```diff
@@ -0,0 +1,126 @@
+
+    def switcher(self, bot_id, binance_client, amount, what_to_do="waiting", test="0"):
+        """bot function that does all what bot does"""
+        client = helpfull_functions.setup_client("localhost", "8086", "test_bot_logs")
```
Do we need initiate influx db client every time? I mean in every loop iteration of the bot? same with API_URL?

amasend commented on bot_management/bot_body.py:
```diff
@@ -0,0 +1,126 @@
+            }]
+
+        if what_to_do == "invest":
+            if test == 2:
```
Please remove such testing code...

bonbon123123 replied:
Should i rewrite 3/5 tests?

edit:
I typed 4, but the intedned number was 3 to begin with.

amasend commented on bot_management/bot_body.py:
```diff
@@ -0,0 +1,126 @@
+            else:
+                try:
+                    buy_order = binance_client.create_order(
+                        symbol='BTCUSDT',
```
I can see a hardcoded symbol here. Why do we have such things hardcoded? User should have elasticity to define what he/she wants.

amasend commented on bot_management/bot_body.py:
```diff
@@ -0,0 +1,126 @@
+                        quantity=amount)
+                    client.write_points(json_body)
+                except BinanceAPIException as e:
+                    # error handling goes here
```
Where is error handling?

amasend commented on bot_management/bot_body.py:
```diff
@@ -0,0 +1,126 @@
+
+    def start_bot(self, api_key, api_secret, test):
+        """bot function that loops it's work till told otherwise(not implemented yet)"""
+        if test == 1:
```
Another test code...

amasend commented on bot_management/bot_body.py:
```diff
@@ -0,0 +1,126 @@
+        else:
+            binance_client = Client(api_key, api_secret)
+            while self.running:
+                print(self.bot_id)
```
We should not accept ay print in a production, better idea will be to implement a logging system but there is no time for that.

bonbon123123 replied:
It is a leftover, didn't notice before commiting.

amasend commented on bot_management/bot_body.py:
```diff
@@ -0,0 +1,126 @@
+                print(self.bot_id)
+                amount_from_AI = 1
+                command_from_AI = "invest"
+                Bot.switcher(self, self.bot_id, binance_client, amount_from_AI, command_from_AI)
```
Now I've just realised that we use a `switcher` name. I cannot instatly tell what this method does. Maybe propose some better naming convention?

amasend commented on bot_management/bot_body.py:
```diff
@@ -0,0 +1,126 @@
+                sleep(10)
+
+
+# on the spot starting bot for testing
```
Not necessary comments here.

amasend commented on bot_management/bot_manager.py:
```diff
@@ -0,0 +1,32 @@
+import uuid
+from bot_management.bot_body import Bot
+import threading
+from bot_management.my_keys import *
```
It is not recommended to user implicite imports. Python should be explicite not implicite.
Also why do we store api_keys in a codebase? These information should be handled by REST API and Postgres.

bonbon123123 replied:
It is just for my testing, so i can store keys in separate file when presenting the code. It is reccomendet to not share your keys.

amasend commented on bot_management/bot_manager.py:
```diff
@@ -0,0 +1,32 @@
+import threading
+from bot_management.my_keys import *
+
+my_array = []
```
What does `my_array` mean? I know this is for bot storage but why this is a somehow 'global' variable not specific for BotManager class/object?

bonbon123123 replied:
Because it worked just as i expected it to work when i put it here and didn,t work when put in other place

amasend commented on bot_management/bot_manager.py:
```diff
@@ -0,0 +1,32 @@
+            if my_array[x].bot_id == bot_id:
+                my_array[x].running = False
+
+# fast example of usage
```
Remove comments.

amasend commented on bot_management/my_keys.py:
```diff
@@ -0,0 +1,19 @@
+from binance.client import Client
+
```
As mentioned earlier, we should not store any apikey in our codebase.

bonbon123123 replied:
it isn't used bu users. only used in tests ant to se if code works before i can connect to rest api

amasend reviewed (changes requested):
In addition to my comments:
- please make better descriptions of methods/functions
- we are missing python annotations
- read how to create python docstrings and follow best practice

bonbon123123 committed:
addresed most of the issues, that were brought up by Amadeusz

Expanded comments, deleted unnecessary comments, changed tests, chaned the location of my_keys.

bonbon123123 committed:
Merge branch 'dev' into trading_bot

bonbon123123 committed:
Merge branch 'dev' into trading_bot
